File: Native/iPhone/Property Finder/Model/Property.m
//
//  Property.m
//  Property Finder
//
//  Created by Colin Eberhardt on 11/10/2012.
//  Copyright (c) 2012 Colin Eberhardt. All rights reserved.
//

#import "Property.h"
#import "NSDictionary+DictionaryJSONHelpers.h"

@implementation Property

- (NSString *)shortTitle
{    
    NSArray* parts = [self.title componentsSeparatedByString:@","];
    if (parts.count >= 2)
    {
        return [NSString stringWithFormat:@"%@/%@", parts[0], parts[1]];
    }
    else
    {
        return self.title;
    }    
}

- (NSString *)formattedPrice
{
    NSNumberFormatter *numberFormat = [[NSNumberFormatter alloc] init];
    [numberFormat setNumberStyle:NSNumberFormatterCurrencyStyle];
    [numberFormat setCurrencySymbol:@"£"];
    [numberFormat setMaximumFractionDigits:0];
    return [numberFormat stringFromNumber:self.price];
}

- (NSString *)bedBathroomText
{
    return [NSString stringWithFormat:@"%d bed, %d bathroom",
            [self.bedrooms intValue], [self.bathrooms intValue]];
}

- (void)toFavouritePropertyDataEntity:(FavouritePropertyDataEntity *)entity
{
    entity.price = self.price;
    entity.bedrooms = self.bedrooms;
    entity.bathrooms = self.bathrooms;
    entity.propertyType = self.propertyType;
    entity.title = self.title;
    entity.summary = self.summary;
    entity.thumbnailUrl = self.thumbnailUrl;
    entity.imageUrl = self.imageUrl;
    entity.guid = self.guid;
}


+ (id)propertyFromJSONData:(NSDictionary *)data
{
    Property* property = [[Property alloc] init];
    
    property.guid = [data NSStringForKey:@"guid"];
    property.price = [data NSNumberForKey:@"price"];
    property.propertyType = [data NSStringForKey:@"property_type"];
    property.bedrooms = [data NSNumberForKey:@"bedroom_number"];
    property.bathrooms = [data NSNumberForKey:@"bathroom_number"];
    property.title = [data NSStringForKey:@"title"];
    property.thumbnailUrl = [data NSStringForKey:@"thumb_url"];
    property.imageUrl = [data NSStringForKey:@"img_url"];
    property.summary = [data NSStringForKey:@"summary"];
    
    return property;
}

+ (id)propertyFromFavouritePropertyDataEntity:(FavouritePropertyDataEntity *)entity
{
    Property* property = [[self alloc] init];
    
    property.price = entity.price;
    property.bedrooms = entity.bedrooms;
    property.bathrooms = entity.bathrooms;
    property.propertyType = entity.propertyType;
    property.title = entity.title;
    property.summary = entity.summary;
    property.thumbnailUrl = entity.thumbnailUrl;
    property.imageUrl = entity.imageUrl;
    property.guid = entity.guid;
    
    return  property;
}



@end

File: xamarinmvvmcross/PropertyCross.Core/Domain/Model/Property.cs
using Newtonsoft.Json.Linq;

namespace PropertyCross.Core.Domain.Model
{
  public class Property
  {
    public Property()
    {
    }

      public Property(JToken jsonProperty)
      {
          Guid = (string) jsonProperty["guid"];
          Price = (int) jsonProperty["price"];
          PropertyType = (string) jsonProperty["property_type"];
          Bedrooms = 0;
          int? bedrooms = jsonProperty.ToNullableInt("bedroom_number");
          if (bedrooms.HasValue)
          {
              Bedrooms = bedrooms.Value;
          }
          Bathrooms = 0;
          int? bathrooms = jsonProperty.ToNullableInt("bathroom_number");
          if (bathrooms.HasValue)
          {
              Bathrooms = bathrooms.Value;
          }
          Title = (string) jsonProperty["title"];
          ThumbnailUrl = (string) jsonProperty["thumb_url"];
          ImageUrl = (string) jsonProperty["img_url"];
          Summary = (string) jsonProperty["summary"];
      }

      public string Guid { get; set;}

    public string Summary { get; set; }

    public int Price { get; set; }

    public int Bedrooms { get; set; }

    public int Bathrooms { get; set; }

    public string PropertyType { get; set; }

    public string Title { get; set; }

    public string ThumbnailUrl { get; set; }

    public string ImageUrl { get; set; }

    public string BedBathroomText
    {
      get
      {
        return string.Format("{0:d} bed, {1:d} bathroom", Bedrooms, Bathrooms);
      }
    }
    public string FormattedPrice
    {
      get
      {
        return string.Format("£{0:0,0}", Price);
      }
    }

    public string ShortTitle
    {
      get
      {
        var titleParts = Title.Split(',');
        if (titleParts.Length >= 2)
        {
          return titleParts[0] + ", " + titleParts[1];
        }
        return Title;
      }
    }
  }
}
